### PR TITLE
Read default regexes from configmap for tests.

### PR DESCRIPTION
**Pull request by abutcher**
@yithian @gregsheremeta 
/hold

### CODE REVIEW

File: pkg/controller/clusterprovision/installlogmonitor_test.go [hunk 1]
```diff
@@ -324,62 +298,16 @@ func TestParseInstallLog(t *testing.T) {
...
-  searchRegexStrings:
-  - "Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane"
-  installFailingReason: KubeAPIWaitFailed
-  installFailingMessage: Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane
-`,
-		},
+// buildRegexConfigMap reads the install log regexes configmap from within config/configmaps/install-log-regexes-configmap.yaml
+func buildRegexConfigMap() runtime.Object {
+	decode := serializer.NewCodecFactory(scheme.Scheme).UniversalDeserializer().Decode
+	stream, err := ioutil.ReadFile("../../../config/configmaps/install-log-regexes-configmap.yaml")
```
Comment (2uasimojo) [inline pkg/controller/clusterprovision/installlogmonitor_test.go:304]: Let's bindata this puppy.

Comment (2uasimojo) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: Oh, look, it [already is](https://github.com/openshift/hive/blob/c0e9a5bfbb68a790101c2f56e1fc40372f4bf102/pkg/operator/assets/bindata.go#L1559).

Comment (2uasimojo) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: To be clear, I'm asking you to use the bindata asset instead of loading up the file "by hand" here.

Comment (abhinavdahiya) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: bindata means we need to make update every time and it's not always clear why my changes come up. since go test runs from the source code directly, it's imo better to use the file directly.

Comment (2uasimojo) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: Our `verify` CI test already enforces that bindata is in sync.

This is what bindata is for.

Comment (abutcher) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: I could see the workflow of adding a new regex/test being a little confusing if you have to make bindata before running this test specifically.

Comment (2uasimojo) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:304]: Fair enough.

File: pkg/controller/clusterprovision/installlogmonitor_test.go [hunk 2]
```diff
@@ -265,43 +267,15 @@ func TestParseInstallLog(t *testing.T) {
...
-  searchRegexStrings:
-  - "compute\\.googleapis\\.com/cpus is not available in [a-z0-9-]* because the required number of resources \\([0-9]*\\) is more than remaining quota"
-  installFailingReason: GCPComputeQuotaExceeded
-  installFailingMessage: GCP CPUs quota exceeded
-`,
-				},
-			}},
+			name:           "GCP compute quota",
+			log:            pointer.StringPtr(gcpCPUQuotaLog),
+			existing:       []runtime.Object{buildRegexConfigMap()},
```
Comment (2uasimojo) [inline pkg/controller/clusterprovision/installlogmonitor_test.go:272]: This is latent, so could be handled in a separate PR, but is there any reason to build this on every test case? It's read-only, right?

**Review by 2uasimojo — COMMENTED** [review]
There are several other test cases with inlined configmaps. If those regexes aren't part of the official config, do we still need to test them?

File: pkg/controller/clusterprovision/installlogmonitor_test.go (hunk 2 above)
Comment (abutcher) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:272]: There are some test cases that want a missing config or a malformed regex so that's why I have it per test case atm.

Comment (abutcher) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:272]: Could split them in a separate PR

**Comment by abutcher** [comment]
If this goes before https://github.com/openshift/hive/pull/1539 @yithian will have to rebase again so
/hold

File: pkg/controller/clusterprovision/installlogmonitor_test.go (hunk 2 above)
Comment (2uasimojo) [reply pkg/controller/clusterprovision/installlogmonitor_test.go:272]: > some test cases that want a missing config or a malformed regex

So there are two different issues here.
1. This comment was about all test cases where `existing` is simply using `buildRegexConfigMap()`. All of those test cases should be able to reuse a common variable built once at the top of the function. That's the bit I'm saying was latent and could be handled in a separate PR.
2. Some of the test cases don't use `buildRegexConfigMap()` but instead use inlined configmaps. Are all of those specifically testing edge cases with missing/bad configmaps?

**Review by 2uasimojo — COMMENTED** [review]
In any case, this is a good start. We can address further cleanup in subsequent PRs.

/lgtm
